src/query-executor/query-executor-base.ts

```diff
@@ -65,7 +68,13 @@ export abstract class QueryExecutorBase implements QueryExecutor {
   ): Promise<QueryResult<R>> {
     return await this.provideConnection(async (connection) => {
       const result = await connection.executeQuery(compiledQuery)
-      return this.#transformResult(result, queryId)
+
+      const transformedResult = await this.#transformResult(result, queryId)
+
+      // TODO: remove.
+      warnOfDeprecatedDriverOrPlugins(result, transformedResult)
```

Comment:
I think we should add a helper function like `logDeprected` (or something) in its own module that keeps a `Set<string>` of deprection messages that have been shown.